ng-dev/auth/shared/firebase.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  GoogleAuthProvider,
+  getAuth,
+  signInWithCredential,
+  linkWithCredential,
+  GithubAuthProvider,
+} from 'firebase/auth';
+import {
+  doTheOAuthDanceWithLoopback,
+  doTheOAuthDanceWithOob,
+  GithubOAuthDanceConfig,
+  GoogleOAuthDanceConfig,
+} from './oauth.js';
+
+export async function loginToFirebase() {
+  /** The type of OAuth dance to do based on whether a session display is available. */
+  const doTheOAuthDance = process.env.DISPLAY
+    ? doTheOAuthDanceWithLoopback
+    : doTheOAuthDanceWithOob;
```

Comment:
Can we use more obvious/official terminology here? e.g. `Device Code Grant" and "Authorization Code" . Loopback and OOB (the abbreviation is pretty non-obvious), seems unclear/non-official? do you know where you got the from?

https://oauth.net/2/grant-types/device-code/